### PR TITLE
Linking cancellationToken with explicitly passed token, Adding ConfigureAwait(false)

### PR DESCRIPTION
**Pull request by Schmaga**
Should resolve all discussed optimizations in #1

### CODE REVIEW

**Comment by georgy93** [comment]
Hi!

In the case where a linked token is created you should dispose of the CancellationTokenSource after finished, in order to avoid memory leaks. The code, however, will become uglier. 
The following can be done with if-else or swich. ( Visual Studio suggested me to use a switch statement)

Example for Send<TResponse>
```   
public async Task<TResponse> Send<TResponse>(IRequest<TResponse> request, CancellationToken cancellationToken = default)
{
            switch (_httpContextAccessor.HttpContext)
            {
                case not null when cancellationToken != default:
                    {
                        using var cts = MergeRequestAbortedCancellationTokenWithDispatchedCancellationToken(cancellationToken);
                        return await _mediator.Send(request, cts.Token).ConfigureAwait(false);
                    }
                case not null when cancellationToken == default:
                    return await _mediator.Send(request, _httpContextAccessor.HttpContext.RequestAborted).ConfigureAwait(false);
                default:
                    return await _mediator.Send(request, cancellationToken).ConfigureAwait(false);
            }
        } 

private CancellationTokenSource MergeRequestAbortedCancellationTokenWithDispatchedCancellationToken(CancellationToken cancellationToken)
             => CancellationTokenSource.CreateLinkedTokenSource(_httpContextAccessor.HttpContext.RequestAborted, cancellationToken);
```

**Comment by Schmaga** [comment]
Should be fixed now

**Comment by Schmaga** [comment]
@georgy93  If you don't have any additional comments, I'm going to merge the state soon and publish a new Nuget package, probably this weekend

File: src/MediatR.Extensions.Microsoft.AspNetCore/Mediator/RequestAbortedCancellationTokenMediatorDecorator.cs [hunk 1]
```diff
@@ -26,34 +29,42 @@ public class RequestAbortedCancellationTokenMediatorDecorator : IMediator
...
         }
 #pragma warning restore 1591
 
-        private CancellationToken GetRequestAbortedCancellationTokenOrUseTokenPassedFromInitialCall(CancellationToken cancellationToken)
+        private CancellationToken DecideCancellationTokenUsage(CancellationToken cancellationToken)
         {
-            return _httpContextAccessor?.HttpContext?.RequestAborted ?? cancellationToken;
+            if (cancellationToken != default && _httpContextAccessor.HttpContext != null)
+            {
+                using var cancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _httpContextAccessor.HttpContext.RequestAborted);
```
Comment (georgy93) [inline src/MediatR.Extensions.Microsoft.AspNetCore/Mediator/RequestAbortedCancellationTokenMediatorDecorator.cs:69]: The CancellationTokenSource will get disposed after exiting the "if" statement, which is before the send/publish command.
Instead, It should be created before the send/publish command and disposed immediately after.
I'm afraid extracting this logic to a method is not an option

**Comment by georgy93** [comment]
Hi!
Sorry for my delayed reply. I'm afraid that the logic for deciding which token to use cannot be extracted in a separate method, because one of the cases requires to create a CancellationTokenSource, which should be created before Send/Publish and disposed immediately after.

**Comment by Schmaga** [comment]
It's a good thing I waited for your reply :) Please check the updated version. In this case it should work now, because the initial call is now wrapped with a potentially newly created token

**Comment by Schmaga** [comment]
I almost forgot to merge this and just remembered. Thanks for your suggestions, I will merge it in the next days.

**Pull request merged**